mne_bids/read.py

```diff
@@ -388,11 +388,22 @@ def read_raw_bids(bids_fname, bids_root, extra_params=None,
 
     # Try to find an associated sidecar.json to get information about the
     # recording snapshot
-    sidecar_fname = _find_matching_sidecar(bids_fname, bids_root,
-                                           '{}.json'.format(kind),
-                                           allow_fail=True)
-    if sidecar_fname is not None:
-        raw = _handle_info_reading(sidecar_fname, raw, verbose=verbose)
+    sidecar_fname = bids_basename + '_{}.json'.format(kind)
+    # Find all potential sidecar files, doing a recursive glob from bids_root
+    search_str = op.join(bids_root, '**', sidecar_fname)
+    candidate_list = glob.glob(search_str, recursive=True)
+    print("Sidecar filnema is: ", sidecar_fname)
+    print(bids_fpath)
+    if len(candidate_list) == 1:
+        # Success
+        sidecar_fname = candidate_list[0]
+
+        if sidecar_fname is not None:
+            raw = _handle_info_reading(sidecar_fname, raw, verbose=verbose)
+    else:
+        warn("All raw datasets should have an accompanying sidecar JSON file."
+             "For the file you want to read in, you should have a file named: "
+             "{}".format(sidecar_fname))
```

Comment:
I think you should modify `_find_matching_sidecar` instead of hacking this outside the function. Also, I don't quite understand the problem. Better to add a test first to demo the problem and then fix it.

Reply:
Okay, so actually the issue was a bit different then what I thought. It seems that if you have another bids - like filename in a sub-directory like ("sourcedata/", or "derivatives"), then this will break because it will find two files. In reality though, it shouldn't recursively search all of bids_root though right? 

It should ignore sourcedata and derivatives. Any suggestions?

Reply:
Just create a copy of the JSON file into "derivatives/" folder or "sourcedata/" and it'll prevent the "sidecars" from being found. 

Can `_find_matching_sidecar` either: recursively only look within "subject" directories, or ignore a handful of keyword directories?

Reply:
Usually such datasets have a `.bidsignore` file, perhaps we should start using that in `mne-bids` too?

Reply:
Would it be sufficient to just add in lines for `_find_matching_sidecar`:

```
# Find all potential sidecar files, doing a recursive glob from bids_root
    search_str = op.join(bids_root, 'sub-**', search_str + '*' + suffix)
```
since all "sidecars" should live within a "subject directory" named `sub-<id>`?